shim/src/package_manager.rs

```diff
@@ -0,0 +1,86 @@
+use crate::paths::AbsolutePath;
+use anyhow::{anyhow, Result};
+use serde::Deserialize;
+use std::fs;
+use std::path::PathBuf;
+
+#[derive(Debug, Deserialize)]
+struct PnpmWorkspaces {
+    pub packages: Vec<PathBuf>,
+}
+
+#[derive(Debug, Deserialize)]
+struct PackageJsonWorkspaces {
+    pub workspaces: Vec<PathBuf>,
+}
+
+pub enum PackageManager {
+    #[allow(dead_code)]
+    Berry,
+    Npm,
+    Pnpm,
+    #[allow(dead_code)]
+    Pnpm6,
+    #[allow(dead_code)]
+    Yarn,
+}
+
+impl PackageManager {
+    /// Returns a list of globs for the package workspace.
+    /// NOTE: We return a `Vec<PathBuf>` instead of a `GlobSet` because we
+    /// may need to iterate through these globs and a `GlobSet` doesn't allow that.
+    ///
+    /// # Arguments
+    ///
+    /// * `root_path`:
+    ///
+    /// returns: Result<Vec<PathBuf, Global>, Error>
+    ///
+    /// # Examples
+    ///
+    /// ```
+    ///
+    /// ```
+    pub fn get_workspace_globs(&self, root_path: &AbsolutePath) -> Result<Vec<PathBuf>> {
```

Comment:
At least for `pnpm`, these can be exclusions (leading `!`). Maybe we shouldn't be treating them as `PathBuf` instances just yet?